R/util.R

```diff
@@ -1262,31 +1262,73 @@ do_on_each_group_2 <- function(df, func1, func2, params1 = quote(list()), params
   ret
 }
 
+#' @export
+#' Utility function that categorizes numeric column based on the type argument.
+#' For example, if type argument is aschar, the column value is categorized as character.
+categorize_numeric <- function(x, type = "asnum") {
+  ret <- NULL
+  switch(type,
+     asnum = {
+       ret <- as.numeric(x)
+     },
+     asint = {
+       ret <- as.integer(x)
+     },
+     asintby10 = {
+       ret <- floor(x/10)*10
+     },
+     aschar = {
+       ret <- as.character(x)
+     })
+  ret
+}
+
 #' @export
 extract_from_date <- function(x, type = "fltoyear") {
   ret <- NULL
   switch(type,
     fltoyear = {
       ret <- lubridate::floor_date(x, unit="year")
     },
+    rtoyear = {# This key is required by Exploratory Desktop for Summarize Group Dialog.
```

Comment:
The point I wanted comment here was,
1. This is same as fltoyear, but it is intentional.
2. The reason why Summary Group Dialog use this different name.

Reply:
Updated comment.